src/strongbox/addon.clj

```diff
@@ -189,13 +189,20 @@
   (logging/with-addon {:dirname (-> addon-dir fs/base-name str)}
     (let [install-dir (str (fs/parent addon-dir))
           addon-dirname (str (fs/base-name addon-dir))
-          ;; todo: this sucks. is there another way we can figure out the relationships between addons in an install-dir other than reading them *all* in?
-          all-addon-data (load-all-installed-addons install-dir game-track)]
-      ;; find the first installed addon that has a `:dirname` (or a grouped addon's `:dirname`) matching the given `addon-dirname`.
-      (first (filter (fn [some-addon]
-                       (let [flattened (mapv :dirname (flatten-addon some-addon))]
-                         (not (nil? (some #{addon-dirname} flattened)))))
-                     all-addon-data)))))
+          target-nfo (nfo/read-nfo install-dir addon-dirname)
+          group-id (:group-id target-nfo)
```

Comment:
assumes nfo data exists. previous implementation would find addon even if it were just toc data.

can we fall back to toc data if no nfo?

Reply:
done!